test/test_common/network_utility.cc

```diff
@@ -104,6 +104,11 @@ const std::string addressVersionAsString(const Address::IpVersion version) {
 
 Address::InstanceConstSharedPtr getSomeLoopbackAddress(Address::IpVersion version) {
   if (version == Address::IpVersion::v4) {
+#ifdef __APPLE__
```

Comment:
One aspect of the rationale here was we would reduce the likelihood of port collision on the few tests that don't bind to port zero and hold the socket, but instead release immediately and expect the port not to be taken by someone else. It's not really helpful in IPv6 though and doesn't fix the underlying issue.

@jamessynge is probably the best person to comment on this.